src/plugins/advanced_settings/public/management_app/components/field/field.test.tsx

```diff
@@ -17,6 +17,10 @@ import { notificationServiceMock, docLinksServiceMock } from '../../../../../../
 import { findTestSubject } from '@elastic/eui/lib/test';
 import { Field, getEditableValue } from './field';
 
+jest.mock('../../../../../kibana_react/public/ui_settings/use_ui_setting', () => ({
```

Comment:
What do you think if instead of mocking inner module we do something like: 

```
const shallowWithKibanaContext = (child: JSX.Element) =>
  shallowWithI18nProvider(
    <KibanaContextProvider services={{ uiSettings: uiSettingsServiceMock.createStartContract() }}>
      {child}
    </KibanaContextProvider>
  );

describe('Field', () => {
  Object.keys(settings).forEach((type) => {
    const setting = settings[type];

    describe(`for ${type} setting`, () => {
      it('should render default value if there is no user value set', async () => {
        const component = shallowWithKibanaContext(
          <Field
            setting={setting}
            handleChange={handleChange}
            enableSaving={true}
            toasts={notificationServiceMock.createStartContract().toasts}
            dockLinks={docLinksServiceMock.createStartContract().links}
          />
        );
        expect(component).toMatchSnapshot();
      });
    });
  });
});
```

I think it will be a little bit more elegant